### PR TITLE
completed the full crud functionalities of the job post

### PR DESCRIPTION
**Pull request by uloamaka**

### CODE REVIEW

File: controllers/companies/jobs.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    res.status(200).json(getJobPost);
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({ err: "Failed to fetch this job post" });
+  }
+};
+// Controller to update a job post by ID
+const updateJobPostByID = async (req, res) => {
+  try {
+    const jobId = req.params.id;
```
Comment (hardeymolar) [inline controllers/companies/jobs.js:42]: convert the jobId to string before passing it

File: controllers/companies/jobs.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({ err: "Failed to update job post" });
+  }
+};
+
+// Controller to delete a job post by ID
+const deleteJobPostByID = async (req, res) => {
+  try {
+    const jobId = req.params.id;
```
Comment (hardeymolar) [inline controllers/companies/jobs.js:60]: same here convert the id before passing it

File: controllers/companies/jobs.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  } catch (err) {
+    console.log(err);
+    res.status(500).json({ err: "Failed to create the job post" });
+  }
+};
+//controller for getting all job post
+const getAllJobPost = async (req, res) => {
+  try {
+    const companyId = req.params.userId;
+    const getSavedJobs = await JobPost.find({ company: companyId }); // Get all saved job post from the DB
```
Comment (hardeymolar) [inline controllers/companies/jobs.js:19]: user:companyId is what we have in the database not company:companyId

**Review by hardeymolar — CHANGES_REQUESTED** [review]

**Comment by uloamaka** [comment]
I have corrected the issues @hardeymolar

**Pull request merged**